Index: BerkeleyDB-0.38/BerkeleyDB.xs
===================================================================
--- BerkeleyDB-0.38/BerkeleyDB.xs
+++ BerkeleyDB-0.38/BerkeleyDB.xs	2009-05-20 17:07:30.000000000 +0200
@@ -893,7 +893,7 @@
 
     vsprintf(buffer2, buffer1, args) ;
 
-    croak(buffer2);
+    croak("%s", buffer2);
 
     /* NOTREACHED */
     va_end(args);
@@ -2281,7 +2281,7 @@
 	    if (SvOK(errfile)) {
 	        FILE * ef = GetFILEptr(errfile);
 	    	if (! ef)
-		    croak("Cannot open file ErrFile", Strerror(errno));
+		    croak("Cannot open file ErrFile: %s", Strerror(errno));
 		RETVAL->ErrHandle = newSVsv(errfile) ;
 	    	env->set_errfile(env, ef) ;
 
